usb/core.py

```diff
@@ -756,6 +758,22 @@ def __init__(self, dev, backend):
         else:
             self.port_number = None
 
+        if getattr(desc, 'speed', None):
+            self.iSpeed = int(desc.speed)
+            if self.iSpeed == 1:
+                self.speed = 'low'
+            elif self.iSpeed == 2:
+                self.speed = 'full'
+            elif self.iSpeed == 3:
+                self.speed = 'high'
+            elif self.iSpeed == 4:
+                self.speed = 'super'
+            else:
+                self.speed = 'unknown'
```

Comment:
I think it is better to let only `speed` be a property and instead of hardcoded numbers, it should be use labeled constants.